service/dap/server.go

```diff
@@ -1150,12 +1150,35 @@ func (s *Server) doCommand(command string) {
 	stopped.Body.AllThreadsStopped = true
 
 	if err == nil {
-		stopped.Body.ThreadId = state.SelectedGoroutine.ID
+		if state.SelectedGoroutine != nil {
+			stopped.Body.ThreadId = state.SelectedGoroutine.ID
+		} else {
+			// If there is no selectedGoroutine, get the list of goroutines and select the first one.
+			// TODO(polina): validate the assumption in this code that the first goroutine
+			// is the current one. So far it appears to me that this is always the main goroutine
+			// with id 1.
+			gs, _, err := s.debugger.Goroutines(0, 1)
+			if err != nil {
+				s.log.Error(err)
+			}
+			if len(gs) > 0 {
+				stopped.Body.ThreadId = gs[0].ID
+			}
+		}
+
 		switch command {
 		case api.Next, api.Step, api.StepOut:
 			stopped.Body.Reason = "step"
 		default:
 			stopped.Body.Reason = "breakpoint"
+			if state.CurrentThread.Breakpoint != nil {
```

Comment:
Good point! Using StopReason seems like it'll help make the logic a bit cleaner. I'll definitely add a test for that.

Reply:
I thought the comment for StopReason said it could be any one reason when there are multiple ones (so could be breakpoint or could be next here, no?) Should the panic/fatal check be outside of the switch to make sure it supplies panic/fatal reason regardless of StopReason?

Reply:
@polinasok yes, that's probably better. Also I suspect that this is the cause of the test failure on Go 1.13 that Travis-CI is reporting.

Reply:
I tried moving the check outside of the switch, but when running with Go 1.13.15, none of the goroutines in the debug state are stopped on a breakpoint when it hits the panic during next.

This is because the breakpoint that is hit is an internal NextDeferBreakpoint, which is cleared before we get to that part of the code. This breakpoint gets set in 1.13 because the "_defer" var is actually pointing to an address in the runtime package (defer at src/runtime/proc.go:157) as opposed to in 1.14 and 1.15 where _defer points to 0. The solution may be to run a different test here (or a separate test for 1.13?), since the StopReasons do appear to be correct.

I can definitely move the checks for these breakpoints outside of the switch to be safe, but it would not fix this breaking test.

Reply:
Makes sense. I'd move the check outside and skip the test on 1.13.